core/src/main/java/org/fao/geonet/kernel/security/openidconnect/SimpleOidcUser.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright (C) 2022 Food and Agriculture Organization of the
+ * United Nations (FAO-UN), United Nations World Food Programme (WFP)
+ * and United Nations Environment Programme (UNEP)
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or (at
+ * your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ *
+ * Contact: Jeroen Ticheler - FAO - Viale delle Terme di Caracalla 2,
+ * Rome - Italy. email: geonetwork@osgeo.org
+ */
+package org.fao.geonet.kernel.security.openidconnect;
+
+import org.fao.geonet.domain.Address;
+import org.fao.geonet.domain.Profile;
+import org.fao.geonet.domain.User;
+import org.springframework.security.oauth2.core.oidc.OidcIdToken;
+import org.springframework.security.oauth2.core.oidc.StandardClaimNames;
+import org.springframework.util.StringUtils;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Contains information about the user (from the oidc ID Token).
+ * <p>
+ * This class knows how to do two things;
+ * a) extra info from the OIDC ID Token
+ * b) move info to a GN user
+ * <p>
+ * parts taken from BaselUser (GN keycloak plugin)
+ */
+class SimpleOidcUser {
+
+
+    private String username;
+    private String firstname;
+    private String surname;
+    private String organisation;
+    private String profile;
+    private String email;
+    private Address address;
+
+
+    SimpleOidcUser(OIDCConfiguration oidcConfiguration, OIDCRoleProcessor oidcRoleProcessor, OidcIdToken idToken) {
+        username = idToken.getPreferredUsername();
+        if (username == null) {
+            username = idToken.getFullName();
+        }
+        if (username != null) {
+            username = org.apache.commons.lang.StringUtils.left(username, 256); //first max 256 chars
+        }
+
+        if (username != null && username.length() > 0) {
```

Comment:
Change to `!org.apache.commons.lang.StringUtils.isBlank(username)`?

Reply:
changed (new commit)

Reply:
changed - new commit.